File: src/main/java/net/atos/zac/app/zaken/model/RESTZaaktype.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.zaken.model;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

import jakarta.validation.constraints.NotNull;

import net.atos.client.zgw.ztc.model.generated.ZaakType;
import net.atos.zac.app.admin.model.RESTZaakafhandelParameters;

public class RESTZaaktype {

  @NotNull public UUID uuid;

  public String identificatie;

  public String doel;

  public String omschrijving;

  public String referentieproces;

  public boolean servicenorm;

  public LocalDate versiedatum;

  public LocalDate beginGeldigheid;

  public LocalDate eindeGeldigheid;

  public ZaakType.VertrouwelijkheidaanduidingEnum vertrouwelijkheidaanduiding;

  public boolean nuGeldig;

  public boolean opschortingMogelijk;

  public boolean verlengingMogelijk;

  public Integer verlengingstermijn;

  public List<RESTZaaktypeRelatie> zaaktypeRelaties;

  public List<UUID> informatieobjecttypes;

  public RESTZaakafhandelParameters zaakafhandelparameters;
}

File: src/main/java/net/atos/zac/app/zaken/model/RESTZaaktype.kt
/*
 * SPDX-FileCopyrightText: 2021 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.zaken.model

import jakarta.validation.constraints.NotNull
import net.atos.client.zgw.ztc.model.generated.ZaakType
import net.atos.zac.app.admin.model.RESTZaakafhandelParameters
import java.time.LocalDate
import java.util.*

class RESTZaaktype {
    @NotNull
    var uuid: UUID? = null

    var identificatie: String? = null

    var doel: String? = null

    var omschrijving: String? = null

    var referentieproces: String? = null

    var servicenorm: Boolean = false

    var versiedatum: LocalDate? = null

    var beginGeldigheid: LocalDate? = null

    var eindeGeldigheid: LocalDate? = null

    var vertrouwelijkheidaanduiding: ZaakType.VertrouwelijkheidaanduidingEnum? = null

    var nuGeldig: Boolean = false

    var opschortingMogelijk: Boolean = false

    var verlengingMogelijk: Boolean = false

    var verlengingstermijn: Int? = null

    var zaaktypeRelaties: List<RESTZaaktypeRelatie>? = null

    var informatieobjecttypes: List<UUID>? = null

    var zaakafhandelparameters: RESTZaakafhandelParameters? = null
}
